common/src/main/java/com/wynntils/features/EmeraldPouchHotkeyFeature.java

```diff
@@ -0,0 +1,197 @@
+import com.wynntils.wc.utils.WynnUtils;
+import it.unimi.dsi.fastutil.ints.Int2ObjectMap;
+import it.unimi.dsi.fastutil.ints.Int2ObjectOpenHashMap;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import net.minecraft.ChatFormatting;
+import net.minecraft.network.chat.MutableComponent;
+import net.minecraft.network.chat.TranslatableComponent;
+import net.minecraft.network.protocol.game.ServerboundContainerClickPacket;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.inventory.ClickType;
+import net.minecraft.world.item.ItemStack;
+import org.lwjgl.glfw.GLFW;
+
+@FeatureInfo(stability = Stability.STABLE, gameplay = GameplayImpact.MEDIUM, performance = PerformanceImpact.SMALL)
+public class EmeraldPouchHotkeyFeature extends Feature {
+
+    private static final Pattern POUCH_USAGE_PATTERN =
+            Pattern.compile("§6§l(\\d* ?\\d* ?\\d*)" + EmeraldSymbols.E_STRING);
+    private static final Pattern POUCH_CAPACITY_PATTERN =
+            Pattern.compile("\\((\\d+)(" + EmeraldSymbols.BLOCKS + "|" + EmeraldSymbols.LE + "|stx) Total\\)");
+
+    // TODO: change sendMessageToClient to GameUpdateOverlay messages once that's available
+    private final KeyHolder emeraldPouchKeybind =
+            new KeyHolder("Open Emerald Pouch", GLFW.GLFW_KEY_UNKNOWN, "Wynntils", true, () -> {
+                if (!WynnUtils.onWorld()) return;
+
+                Player player = McUtils.player();
+                HashMap<Integer, Pair<ItemStack, Pair<Integer, Integer>>> emeraldPouches = new HashMap<>() {};
+
+                for (int i = 0; i < player.getInventory().getContainerSize(); i++) {
+                    ItemStack stack = player.getInventory().getItem(i);
+                    if (!stack.isEmpty() && WynnItemMatchers.isEmeraldPouch(stack)) {
+                        emeraldPouches.put(
+                                i, new Pair<>(stack, new Pair<>(getPouchUsage(stack), getPouchCapacity(stack))));
+                    }
+                }
+                Int2ObjectMap<ItemStack> changedSlots = new Int2ObjectOpenHashMap<>();
+
+                pouchSwitch:
+                switch (emeraldPouches.size()) {
+                    default -> {
+                        Integer slotNumber = -1;
+                        ItemStack slotStack = null;
+
+                        boolean alreadyHasNonEmpty = false;
+                        Integer usedPouch = -1;
+                        ItemStack pouchStack = null;
+                        for (Integer key : emeraldPouches.keySet()) {
+                            if (emeraldPouches.get(key).b.a > 0
+                                    && !alreadyHasNonEmpty) { // Found one pouch with a nonzero balance, remember this
+                                alreadyHasNonEmpty = true;
+                                usedPouch = key; // Save our pouch slot ID
+                                pouchStack = emeraldPouches.get(key).a;
```

Comment:
usedPouch and pouchStack are only used once, which allows for the replacement
```java
slotNumber = key; // Save our pouch slot ID
slotStack = emeraldPouches.get(key).a;
```
without saving